docs/coding/recipes/balance-bounds.md

```diff
@@ -0,0 +1,103 @@
+---
+sidebar_position: 5
+---
+
+# Balance Bounds
+
+It is easy to limit an account's balance using either
+[`flags.debits_must_not_exceed_credits`](../../reference/account.md#flagsdebits_must_not_exceed_credits)
+or
+[`flags.credits_must_not_exceed_debits`](../../reference/account.md#flagscredits_must_not_exceed_debits).
+
+What if you want an account's balance to stay between an upper and a lower bound?
+
+This is possible to check atomically using a set of linked transfers.
+
+## Preconditions
+
+1. Target Account Should Have a Limited Balance
+
+The account whose balance you want to bound should have one of these flags set:
+
+- [`flags.debits_must_not_exceed_credits`](../../reference/account.md#flagsdebits_must_not_exceed_credits)
+  for accounts with [credit balances](../data-modeling.md#credit-balances)
+- [`flags.credits_must_not_exceed_debits`](../../reference/account.md#flagscredits_must_not_exceed_debits)
+  for accounts with [debit balances](../data-modeling.md#debit-balances)
+
+2. Create a Control Account with the Opposite Limit
+
+There must also be a designated control account.
+
+As you can see below, this account will never actually take control of the target account's funds,
+but we will set up simultaneous transfers in and out of the control account to apply the limit.
+
+This account must have the opposite limit applied as the target account:
+
+- [`flags.credits_must_not_exceed_debits`](../../reference/account.md#flagscredits_must_not_exceed_debits)
+  if the target account has a [credit balance](../data-modeling.md#credit-balances)
+- [`flags.debits_must_not_exceed_credits`](../../reference/account.md#flagsdebits_must_not_exceed_credits)
+  if the target account has a [debit balance](../data-modeling.md#debit-balances)
+
+3. Create an Operator Account
+
+The operator account will be used to fund the Control Account.
+
+## Executing a Transfer with a Balance Bounds Check
+
+This consists of 5 [linked transfers](../../reference/requests/README.md#linked-events).
+
+We will refer to two amounts:
+
+- The **limit amount** is upper bound we want to maintain on the target account's balance.
+- The **transfer amount** is the amount we want to transfer if and only if the target account's
+  balance is within the bounds.
```

Comment:
`if and only if the target account's balance after a successful transfer would be within the bounds`